bootstrap fails with pip 18.0 installed

Following crash when running bootstrap with ```--download-only``` on pip 18.
Works fine on pip <= 10.
Solution appears to be to override pip requirements tracker location.

```
  Downloading https://files.pythonhosted.org/packages/00/2b/8d082ddfed935f3608cc61140df6dcbf0edea1bc3ab52fb6c29ae3e81e85/future-0.16.0.tar.gz (824kB)
    100% |████████████████████████████████| 829kB 24.0MB/s 
  Saved ./future-0.16.0.tar.gz
Exception:
Traceback (most recent call last):
  File "/scratch/jenkins_slave/workspace/dials_bootstrap_prerequisites/python27/lib/python2.7/site-packages/pip/_internal/basecommand.py", line 141, in main
    status = self.run(options, args)
  File "/scratch/jenkins_slave/workspace/dials_bootstrap_prerequisites/python27/lib/python2.7/site-packages/pip/_internal/commands/download.py", line 224, in run
    resolver.resolve(requirement_set)
  File "/scratch/jenkins_slave/workspace/dials_bootstrap_prerequisites/python27/lib/python2.7/site-packages/pip/_internal/resolve.py", line 102, in resolve
    self._resolve_one(requirement_set, req)
  File "/scratch/jenkins_slave/workspace/dials_bootstrap_prerequisites/python27/lib/python2.7/site-packages/pip/_internal/resolve.py", line 256, in _resolve_one
    abstract_dist = self._get_abstract_dist_for(req_to_install)
  File "/scratch/jenkins_slave/workspace/dials_bootstrap_prerequisites/python27/lib/python2.7/site-packages/pip/_internal/resolve.py", line 209, in _get_abstract_dist_for
    self.require_hashes
  File "/scratch/jenkins_slave/workspace/dials_bootstrap_prerequisites/python27/lib/python2.7/site-packages/pip/_internal/operations/prepare.py", line 297, in prepare_linked_requirement
    with self.req_tracker.track(req):
  File "/dls_sw/prod/tools/RHEL6-x86_64/Python/2-7-3/prefix/lib/python2.7/contextlib.py", line 17, in __enter__
    return self.gen.next()
  File "/scratch/jenkins_slave/workspace/dials_bootstrap_prerequisites/python27/lib/python2.7/site-packages/pip/_internal/req/req_tracker.py", line 74, in track
    self.add(req)
  File "/scratch/jenkins_slave/workspace/dials_bootstrap_prerequisites/python27/lib/python2.7/site-packages/pip/_internal/req/req_tracker.py", line 51, in add
    with open(entry_path, 'w') as fp:
IOError: [Errno 2] No such file or directory: '/tmp/pip-req-tracker-RdUIUW/93b16de836b6e2be5728a20570c7619783f8611cb14149e656816433'
Traceback (most recent call last):
  File "modules/cctbx_project/libtbx/auto_build/install_base_packages.py", line 1648, in <module>
    installer(args=sys.argv, log=sys.stdout)
  File "modules/cctbx_project/libtbx/auto_build/install_base_packages.py", line 247, in __init__
    self.build_dependencies(packages=packages)
  File "modules/cctbx_project/libtbx/auto_build/install_base_packages.py", line 764, in build_dependencies
    getattr(self, 'build_%s'%i)()
  File "modules/cctbx_project/libtbx/auto_build/install_base_packages.py", line 958, in build_python_compatibility
    confirm_import_module='future')
  File "modules/cctbx_project/libtbx/auto_build/install_base_packages.py", line 662, in build_python_module_pip
    assert pip_call(pip_cmd) == 0, 'pip download failed'
AssertionError: pip download failed
Process failed with return code 1
```

--- a/libtbx/auto_build/install_base_packages.py
+++ b/libtbx/auto_build/install_base_packages.py
@@ -658,6 +658,7 @@ Installation of Python packages may fail.
         pip_cmd.append('--no-cache-dir')
       else:
         pip_call = pip.main
+      os.environ['PIP_REQ_TRACKER'] = pkg_info['cachedir']
       print "  Running with pip:", pip_cmd
       assert pip_call(pip_cmd) == 0, 'pip download failed'
       return

